library/init/algebra/functions.lean

```diff
@@ -24,7 +24,9 @@ solve1 $ intros
 >> try `[apply le_of_not_le, assumption]
 
 meta def tactic.interactive.min_tac (a b : interactive.parse lean.parser.pexpr) : tactic unit :=
-`[by_cases (%%a ≤ %%b), repeat {min_tac_step}]
+interactive.by_cases (none, ``(%%a ≤ %%b)); min_tac_step
+-- TODO(Mario): why doesn't this work?
+--`[by_cases (%%a ≤ %%b), repeat {min_tac_step}]
```

Comment:
@gebner Do you know why the `expr.subst` application is not well typed here? From the output term, it seems like the modified `by_cases` parser is still able to correctly deduce the structure even though there are holes in the preterm, but the output term says it should be a `bool` instead of a `pexpr` which doesn't make any sense.

Reply:
Ah, you've uncovered a dirty hack of mine: The instance `has_reflect pexpr` is a lie for preterms with antiquotations because, being outside of `tactic`, it cannot elaborate the antiquotations nor synthesize `to_pexpr` instances for them. Instead, for interactive parameters of type `pexpr`, we hackishly treat the reflected expression as a preterm after all: https://github.com/leanprover/lean/blob/d83b9ef3efbf2d9eafce32c26d047e1f69539a2f/src/library/vm/vm_parser.cpp#L69. Because you're returning a `pexpr` in a tuple, this doesn't trigger.

Really, what we should do is changing `parse` from taking a `has_reflect` instance to some new typeclass `has_prereflect` that produces preterms. It would not be a big change (we can give a transitive instance from `has_reflect`), but I'm not sure it would make sense to tackle this before #1674.